core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -704,6 +704,11 @@ default boolean isMaterializedView(Session session, QualifiedObjectName viewName
      */
     boolean isValidTableVersion(Session session, QualifiedObjectName tableName, TableVersion version);
 
+    /**
+     * Returns true if the connector reports number of written bytes. Otherwise, it returns false.
+     */
+    boolean supportsReportingWrittenBytes(Session session, TableMetadata tableMetadata);
```

Comment:
TableMetadata is a container object that describes the shape of a table (columns, properties, etc). Using it to *identify* a table is not appropriate, especially given that a caller can make up any TableMetadata it wants.

 Tables should be identified either by name (catalog/schema/name) or by handle (i.e., and opaque identifier).

Reply:
@martint, I cannot use a handle because this kind of object represents existing table. I can use the full name - no problem. If rewrite the code to use name instead of `TableMetadata`, will be that PR acceptable for you?

Reply:
ok @martint we go with that

Reply:
@arhimondr @radek-starburst @martint 

i happened to discuss this with @radek-starburst today

- for existing tables (INSERT) the `ConnectorTableHandle` should be presented to the `ConnectorMetadata` to drive the decision. `ConnectorTableHandle` is the best info we can provide, so let's use it,
- for new tables (CTAS) the `ConnectorTableMetadata` should be presented to the connector. It contains the best information possible. Showing only `properties` is not sufficient -- it allows connector to inspect eg partitioning, but eg doesn't allow to inspect types.
- for UPDATE and DELETE nothing is needed since writer scaling is not applicable for these operations -- the operations happen within source split
- for TABLE EXECUTE -- ideally we should provide `ConnectorTableHandle` + `ConnectorTableExecuteHandle`, since connector's decision may depend on the actual operation being executed. However, since we don't have such use-case today AND it is easy to backwards-compatibly add a new such API in the future, let's provide `ConnectorTableHandle` only as in INSERT case